frontend/src/scenes/Auth/sagas/index.ts

```diff
@@ -0,0 +1,60 @@
+import { all, put, call, takeEvery } from 'redux-saga/effects';
+import { Routine } from 'redux-saga-routines';
+import { fetchUserRoutine, addNewUserRoutine, loginUserRoutine } from '../routines';
+import { registration, login, fetchUser } from '../../../services/authService';
+import { IAuthServerResponse } from '../../../common/models/auth/AuthServerResponse';
+import { IUser } from '../../../common/models/user/IUser';
+import { setTokens, clearStorage } from '../../../common/helpers/storageHelper';
+
+function* fetchUserRequest() {
+  try {
+    const user:IUser = yield call(fetchUser);
+    yield put(fetchUserRoutine.success(user));
+  } catch (error) {
+    yield call(clearStorage);
+    yield put(fetchUserRoutine.failure(error.message));
+  }
+}
+
+function* watchUserRequest() {
+  yield takeEvery(fetchUserRoutine.TRIGGER, fetchUserRequest);
+}
+
+function* loginUserRequest({ payload }: Routine<any>) {
+  try {
+    const { accessToken, refreshToken, user }: IAuthServerResponse = yield call(login, payload);
+    setTokens({ accessToken, refreshToken });
+    yield put(loginUserRoutine.success(user));
+  } catch (error) {
+    yield put(loginUserRoutine.failure(error.message));
+  }
+}
+
+function* watchLoginUserRequest() {
+  yield takeEvery(loginUserRoutine.TRIGGER, loginUserRequest);
+}
+
+function* addNewUserRequest({ payload }: any): Routine<any> {
+  try {
+    const { accessToken, refreshToken, user }: IAuthServerResponse = yield call(registration, payload);
+    setTokens({ accessToken, refreshToken });
+    yield put(addNewUserRoutine.success(user));
+  } catch (error) {
+    if (error.message.includes('duplicate key value')) {
```

Comment:
That's an unreliable strategy. What if PostgreSQL is localized and will return errors in a different language? For these reasons we should send CustomError from server with a proper `errorCode`. Error code should be enum. Then you'll be able to check error by its errorCode correctly.